internal/controller/components/ray/ray_controller_actions.go

```diff
@@ -59,3 +63,30 @@ func devFlags(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
 	// TODO: Implement devflags logmode logic
 	return nil
 }
+
+// This function is used to perform the sanity checks for the Ray component upgrade.
+// The configuration is ok if CodeFlare component resource is not present in the cluster,
+// as it is required to remove it before upgrade to ODH v3.
+// If CRD is not present, sanity check passes.
+func performV3UpgradeSanityChecks(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
```

Comment:
nit: probably not for this PR, but this seems to be a good candidate to become a shared gating action

Reply:
Something like: `sanitycheck.NewAction(cluster.HasResource(...))`?

Reply:
something like that